packages/gatsby/index.d.ts

```diff
@@ -394,16 +394,16 @@ export interface GatsbyNode {
   setFieldsOnGraphQLNodeType?(
     args: SetFieldsOnGraphQLNodeTypeArgs,
     options: PluginOptions
-  ): any
+  ): GraphQLFieldConfigMap<any, any>
   setFieldsOnGraphQLNodeType?(
     args: SetFieldsOnGraphQLNodeTypeArgs,
     options: PluginOptions
-  ): Promise<any>
+  ): Promise<GraphQLFieldConfigMap<any, any>>
   setFieldsOnGraphQLNodeType?(
     args: SetFieldsOnGraphQLNodeTypeArgs,
     options: PluginOptions,
     callback: PluginCallback
-  ): void
+  ): GraphQLFieldConfigMap<any, any>
```

Comment:
for `callback` cases - return should be `void` - we instead use results from callback args, so potentially this should be something like this (and `PluginCallback` should be able to handle accepting result types, as it doesn't right now):

```ts
  setFieldsOnGraphQLNodeType?(
    args: SetFieldsOnGraphQLNodeTypeArgs,
    options: PluginOptions,
    callback: PluginCallback<GraphQLFieldConfigMap<any, any>>
  ): void
```

Reply:
Right, thanks, I went ahead and made that return type mandatory to improve readability (`callback?: PluginCallback<void>`).